### PR TITLE
invalid pointer / crash in getotherfuncs()

### PR DESCRIPTION
**Pull request by Osndok**

### CODE REVIEW

File: source/jsDecode.brs [hunk 1]
```diff
@@ -122,6 +122,11 @@ Function getOtherFuncs(primary_func as Object, jsText as String) as Object
     '""" Return all secondary functions used in primary_func. """
     'print("scanning javascript for secondary functions.")
     body = primary_func.body
+    
+    if ( body = invalid ) then
```
Comment (Protuhj) [inline source/jsDecode.brs]: Do you have a video that exhibits this error case?

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,5 @@
 source/.DS_Store
 images/.DS_Store
 updateZip.bat
-den.bat
\ No newline at end of file
+den.bat
+/dist
```
Comment (Protuhj) [inline .gitignore]: I'm not sure where the 'dist' directory comes from. Is that something you've created?

File: source/jsDecode.brs (hunk 1 above)
Comment (Osndok) [inline source/jsDecode.brs]: It was happening to me all the time, particularly when browsing reddit videos. I don't have a particular case at hand, but you might be able to get what you need (or make a better assessment) from the log:

```
Http: # 30324 done status: 200 time: 394ms request: GET http://www.reddit.com/r/videos/top.json?after=t3_41cfag&t=day
Http: # 30325 done status: 200 time: 445ms request: GET http://www.reddit.com/r/videos/top.json?after=t3_41ge7k&t=day
Http: # 30326 done status: 200 time: 470ms request: GET http://www.reddit.com/r/videos/top.json?after=t3_41etyj&t=day
User-Agent      Mozilla/5.0 (Windows NT 6.1; WOW64; rv:28.0) Gecko/20100101 Firefox/28.0
Cookie          
Http: # 30327 done status: 200 time: 260ms request: GET http://www.youtube.com/get_video_info?el=detailpage&video_id=reQmc_9cRvg
User-Agent      Mozilla/5.0 (Windows NT 6.1; WOW64; rv:28.0) Gecko/20100101 Firefox/28.0
Http: # 30328 done status: 200 time: 699ms request: GET http://www.youtube.com/watch?v=reQmc_9cRvg
User-Agent      Mozilla/5.0 (Windows NT 6.1; WOW64; rv:28.0) Gecko/20100101 Firefox/28.0
Http: # 30329 done status: 200 time: 827ms request: GET https://s.ytimg.com/yts/jsbin/player-en_US-vflGR-A-c/base.js
Couldn't find function pr

Current Function:
119:  Function getOtherFuncs(primary_func as Object, jsText as String) as Object
120:      '""" Return all secondary functions used in primary_func. """
121:      'print("scanning javascript for secondary functions.")
122:      body = primary_func.body
123:*     body = body.Tokenize(";")
124:      '# standard function call; X=F(A,B,C...)
125:      funcCall = CreateObject( "roRegex", "(?:[$\w+])=([$\w]+)\(((?:\w+,?)+)\)$", "" )
126:      '# dot notation function call; X=O.F(A,B,C..)
127:      dotCall = CreateObject( "roRegex", "(?:[$\w+]=)?([$\w]+)\.([$\w]+)\(((?:\w+,?)+)\)$", "" )
'Dot' Operator attempted with invalid BrightScript Component or interface reference. (runtime error &hec) in pkg:/source/jsDecode.brs(123)
123:     body = body.Tokenize(";")
Backtrace:
#11 Function getotherfuncs(primary_func As Object, jstext As String) As Object
   file/line: pkg:/source/jsDecode.brs(123)
#10 Function get_js_sm(video_id As Dynamic) As Dynamic
   file/line: pkg:/source/jsDecode.brs(67)
#9  Function getyoutubemp4url(video As Object) As Integer
   file/line: pkg:/source/video.brs(1232)
#8  Function video_get_qualities(video As Object) As Integer
   file/line: pkg:/source/video.brs(1671)
#7  Function videodetails_impl(thevideo As Object, breadcrumb As String, videos As Dynamic, idx As Dynamic) As Integer
   file/line: pkg:/source/video.brs(959)
#6  Function $anon_75(categories As Dynamic, youtube As Dynamic, video As Dynamic, category_idx As Dynamic, set_idx As Dynamic) As Dynamic
   file/line: pkg:/source/reddit.brs(100)
#5  Function uitkdocategorymenu(categorylist As Dynamic, screen As Dynamic, content_callback As Dynamic, onclick_callback As Dynamic, onplay_func As Dynamic, isplaylist As Boolean, isreddit As Boolean) As Integer
   file/line: pkg:/source/uiToolkit.brs(516)
#4  Function viewreddits(youtube As Object) As String
   file/line: pkg:/source/reddit.brs(107)
#3  Function $anon_0(menu As Dynamic, youtube As Dynamic, set_idx As Dynamic) As Dynamic
   file/line: pkg:/source/appMain.brs(96)
#2  Function uitkdopostermenu(posterdata As Dynamic, screen As Dynamic, onselect_callback As Dynamic) As Dynamic
   file/line: pkg:/source/uiToolkit.brs(91)
#1  Function showhomescreen() As Void
   file/line: pkg:/source/appMain.brs(139)
#0  Function runuserinterface() As Void
   file/line: pkg:/source/appMain.brs(32)
Local Variables:
primary_func     roAssociativeArray refcnt=2 count:1
jstext           roString (2.1 was String) refcnt=3 val:"var _yt_player={};(function(g){var window=this;var f,n=function(a){return void 0!==a},r=functi"...
global           rotINTERFACE:ifGlobal
m                roAssociativeArray refcnt=10 count:5
body             Invalid
funccall         <uninitialized>
dotcall          <uninitialized>
functions        <uninitialized>
part             <uninitialized>
match            <uninitialized>
name             <uninitialized>

BrightScript Debugger>
```
Does that help?

Comment (Protuhj) [inline source/jsDecode.brs]: I've seen that error in the past, were you running the latest 'developer version'? (not the full release)

**Pull request closed without merging**